etna/lib/etna/clients/magma/workflows/record_synchronization_workflow.rb

```diff
@@ -30,17 +30,23 @@ def copy_model(model_name, copied_models = Set.new)
 
           descendants = source_models.to_directed_graph.descendants("project")
           (descendants[model_name] || []).each do |required_model|
-            copy_model(required_model, copied_models)
+            copy_model(required_model, copied_models) do |update|
+              yield update if block_given?
+            end
           end
 
-          model = source_client.retrieve(RetrievalRequest.new(project_name: project_name, model_name: model_name, record_names: 'all')).models.model(model_name)
-          return if model.nil?
-
-          crud.update_records do |update_request|
-            model.documents.document_keys.each do |identifier|
-              record = model.documents.document(identifier)
-              record = record.each.select { |k, v| v != nil && !v.is_a?(Array) }.to_h
-              update_request.update_revision(model_name, identifier, record)
+          crud.page_records(model_name) do |documents|
+            yield [model_name, documents] if block_given?
+            begin
+              crud.update_records do |update_request|
+                documents.document_keys.each do |identifier|
+                  record = documents.document(identifier)
+                  record = record.each.select { |k, v| v != nil && !v.is_a?(Array) }.to_h
+                  update_request.update_revision(model_name, identifier, record)
+                end
+              end
+            rescue => e
+              raise unless ignore_update_errors
```

Comment:
When using this workflow, there is no guarantee that existing records match validations; the ability to optionally ignore update errors allowed me to test my etl easier locally by copying the records that pass validations and ignore those that don't.